### PR TITLE
Change primitive handling in RemoveObjectsIsNull

### PR DESCRIPTION
**Pull request by timtebeek**
Fixes https://github.com/openrewrite/rewrite/issues/3915
Fixes https://github.com/openrewrite/rewrite/issues/3916

### CODE REVIEW

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java [hunk 1]
```diff
@@ -63,58 +62,35 @@ public void test() {
...
-            """
-              import static java.util.Objects.nonNull;
+              import static java.util.Objects.isNull;
               public class A {
-                  public void test() {
-                      boolean a = true;
-                      if (a != null) {
-                          System.out.println("a is non-null");
+                  public void test(boolean a) {
+                      if (isNull(a)) {
```
Comment (koppor) [inline rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java:92]: This should be changed, shouldn't it?

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java [hunk 2]
```diff
@@ -125,27 +101,26 @@ public void test() {
 
     @Issue("https://github.com/openrewrite/rewrite/issues/1547")
     @Test
-    void transformCallToNonNullNeedsParentheses() {
```
Comment (koppor) [inline rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java:128]: Was this test completely removed by intention?

**Review by koppor — COMMENTED** [review]
Sponteneous comments just checking the diff.

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java (hunk 1 above)
Comment (timtebeek) [reply rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java:92]: I had my doubts about this case; when we upcast from `boolean` to `Boolean` it doesn't really make sense to even call `isNull` with an upcast to `Boolean` for that method call. Perhaps we should completely remove the method call in those cases?

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java (hunk 2 above)
Comment (timtebeek) [reply rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java:128]: Yes after the recipe changes this was a duplicate. We can revise this of course based on the other feedback here, but up to this commit it was intentional.

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java (hunk 1 above)
Comment (koppor) [reply rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java:92]: Oh, yeah! And replace !isNull by `false` 😅

Comment (timtebeek) [reply rewrite-java-test/src/test/java/org/openrewrite/java/RemoveObjectsIsNullTest.java]: Yup; I've changed the logic now to add simple `true` or `false` instead. [In rewrite-launchdarkly](https://github.com/openrewrite/rewrite-launchdarkly/blob/87279dabe8a7d57260575381f4da40dd2262f251/src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java#L67) we follow such a replacement up with `SimplifyConstantIfBranchExecution`, but that would require this recipe to move to rewrite-static-analysis. 🤔

**Pull request merged**